### PR TITLE
[WIP] Documentation update

### PR DESCRIPTION
**Pull request by mxstbr**
This is an initial WIP of some documentation improvements, especially focussing on removal of stuff (#223, cc @oliverturner) and much more and better explanations on how to use stuff. (esp. the JavaScript documentation was/is still very much lacking)

**I need help with this**, we need short usage and removal guides for almost all our JS features! Jump in and fill out the empty files I left in there.

- [x] Routing
- [x] Reselect
- [x] redux-saga

### CODE REVIEW

**Comment by mxstbr** [comment]
Alright, I finished writing all documentation we need (?) and added ToC to all READMEs for easy navigation. @oliverturner could you take one last look at this and fix/append/remove problems you see?

> Note: I did not add a "Remove routing" section, as the routing is very tightly integrate with this boilerplate and anybody who'd want to remove it but doesn't know how isn't suited to use it anyway.

**Comment by oliverturner** [comment]
On it :)

File: docs/css/css-modules.md [hunk 1]
```diff
@@ -11,8 +15,11 @@ Write your CSS normally in the `styles.css` file in the component folder.
 ```CSS
 /* styles.css */
 
-.myClassName {
-  color: green;
+.saveBtn {
+  composes: btn from '../components/btn'; // Yay for composition!
```
Comment (mxstbr) [inline docs/css/css-modules.md]: :+1:

**Comment by mxstbr** [comment]
:clap: This is great, thanks so much for checking that again – will merge now! Thanks for your help!!

**Comment by kachkaev** [comment]
@mxstbr this might be a not very good place to ask, but I'm not sure the topic deserves a new issues or a pr. What do you think of extending `docs/css/postcss.md` a bit in order to explain how to add your own postcss plugin?

The steps for me included:
* `npm install postcss-position`
* Modification of `webpack.dev.babel.js` and `webpack.prod.babel.js`
  * `const postcssPosition = require('postcss-position');`
  * `postcssPlugins: [ ...` → `postcssPlugins: [ postcssPosition(), ...`
* `Ctrl+C` dev server and `npm start` as webpack does not pick config changes while running.

Can't say I spent a lot of time on figuring all this up, however knowing what to do in advance would be helpful.

Feel free to add the above steps into docs using the language style you prefer and with the amount of detail you find sufficient.

Thanks for the boilerplate! I enjoy it more and more each day!

**Comment by mxstbr** [comment]
Thanks for the compliments @kachkaev, that sounds like a useful addition! Do you want to submit a PR?

**Comment by kachkaev** [comment]
I might have a look at this in a week or so – too busy now. If anyone else wants to volunteer, feel free to use my steps above as a base!

**Pull request merged**